finance.py

```
@@ -241,6 +238,30 @@ def process_data(self):
         self.___undervalued_stocks.sort(key=self.get_sorting_key, reverse=True)
         self.dump_stocks()
 
+    # TODO: ADD DISCOUNT RATE AND GROWTH RATE
+    def database_dcf_run(database_name):
+        file_type = database_name.split(".", 1)[1]
+
+        stock_list = []
+        if file_type == "p": #pickle
+            stock_list = self.load_pickle(database_name)
```

Ruff (0.11.9)

Undefined name `self`

(F821)